### ISSUE TITLE
How to pass query params in resolver to the output object?

### ISSUE THREAD

**Issue by azzamsa** [async-graphql/async-graphql#989]
Hi.

If I have this `users` resolver.

```rust
#[Object]
impl UserQuery {
    pub async fn users(
        &self,
        ctx: &Context<'_>,
        after: Option<String>,
        before: Option<String>,
        first: Option<i32>,
        last: Option<i32>,
    ) -> FieldResult<UserConnection> {
        let server_ctx = ctx.data::<Arc<ServerContext>>()?;

        let result = server_ctx
            .user_service
            .find_users(after.clone(), before.clone(), first, last)
            .await;
        match result {
            Ok(users) => Ok(users),
            Err(err) => Err(Error::new(err.to_string())),
        }
    }
```

How to pass `after, before, first, last` to the output object. in this case `total_count` ?

```rust
#[derive(Debug, Clone)]
pub struct UserConnection {
    // A list of edges.
    pub edges: Vec<UserEdge>,
    // Information to aid in pagination.
    pub page_info: PageInfo,
    // Identifies the total count of items in the connection.
    pub total_count: i64,
}

#[Object]
impl UserConnection {
    pub async fn edges(&self) -> Vec<UserEdge> {
        self.edges.clone()
    }
    pub async fn page_info(&self) -> PageInfo {
        self.page_info.clone()
    }
    pub async fn total_count(&self, ctx: &Context<'_>) -> i64 {

   //
  // How to get these variables?

        let server_ctx = ctx.data::<Arc<ServerContext>>().unwrap();
        let result = server_ctx
            .user_service
            .total_count(after.clone(), before.clone(), first, last)
            .await
            .unwrap();
        result
    }
}
```
The input API should looks like this.

``` graphql
query {
  users(first: 3) {
    edges {
      cursor
      node {
        id
        name
      }
    }
    totalCount
  }
}

```

The params should be in `users()` not in `totalCount()`.

**Comment by sunli829** [comment]
Should use the [query](https://docs.rs/async-graphql/4.0.5/async_graphql/types/connection/fn.query.html) function to create a `Connection`.

**Comment by azzamsa** [comment]
But, is there a way to get query params that is not part of the struct, such as `repeat` here?

``` rust
#[Object]
impl Student {
    async fn name(&self) -> String {
        self.name.to_string()
    }

    async fn repeat_name(&self, ctx: &Context<'_>) -> String {
        //
        //⚠️ how to get `repeat` ?
        //
        let repeat = self.repeat; // ⚠️
        format!("{} ", self.name).repeat(4)
    }
}

//
// Query
//

#[derive(Default)]
pub struct StudentQuery;

#[Object]
impl StudentQuery {
    pub async fn student(&self, _ctx: &Context<'_>, repeat: i8) -> Student {
        Student {
            name: "Ferris".to_string(),
            repeat_name: "unreachable()".to_string(),
        }
    }
}
```

**Comment by sunli829** [comment]
```rust
struct Student {
   repeat: u8,
   ...
}

#[Object]
impl Student {
    async fn name(&self) -> String {
        self.name.to_string()
    }

    async fn repeat_name(&self, ctx: &Context<'_>) -> String {
        //
        //⚠️ how to get `repeat` ?
        //
        let repeat = self.repeat; // ⚠️
        format!("{} ", self.name).repeat(4)
    }
}

//
// Query
//

#[derive(Default)]
pub struct StudentQuery;

#[Object]
impl StudentQuery {
    pub async fn student(&self, _ctx: &Context<'_>, repeat: i8) -> Student {
        Student {
            repeat, // <<<<<<<< pass the repeat to Student 
            name: "Ferris".to_string(),
            repeat_name: "unreachable()".to_string(),
        }
    }
}
```

**Comment by azzamsa** [comment]
So, there is no way to get the params value without it being in the Object struct.
Then my solutions are:

1. Put the `repeat` in the struct, but hide [Hide content in introspection](https://async-graphql.github.io/async-graphql/en/visibility.html?highlight=hide#hide-content-in-introspection).
2. Use another approach with [lookahead](https://docs.rs/async-graphql/latest/async_graphql/struct.Lookahead.html#method.field)
